lib/elixir/test/elixir/stream_test.exs

```diff
@@ -168,6 +168,14 @@ defmodule StreamTest do
     stream = Stream.cycle([1, 2, 3])
     assert is_function(stream)
 
+    assert_raise ArgumentError, fn ->
+      Stream.cycle([])
+    end
+
+    assert_raise ArgumentError, fn ->
+      Stream.map([], &(&1)) |> Stream.cycle |> Enum.to_list
```

Comment:
`%{}` is an empty non-list enumerable as well, I think it would work fine here. Also, please use parens for 0-arity calls as well 👍